app/models/service.rb

```diff
@@ -41,6 +45,10 @@ def add_resource(rsc, options = {})
   end
   alias_method :<<, :add_resource
 
+  def provision_dialog
+    miq_request.try(:provision_dialog)
+  end
```

Comment:
This can be handled with delegate: `delegate :provision_dialog, :to => :miq_request, :allow_nil => true`